scripts/tags/video.js

```diff
@@ -0,0 +1,140 @@
+(function() {
+  'use strict';
+  
+  /**
+   * search index in an array with a regex
+   * @param {Array} array
+   * @param {Regex} regex
+   * @param {Number} startpos
+   * @return {Number}
+   */
+
+  var rPath = new RegExp(
+    '((([A-Za-z]{3,9}:(?:\\/\\/)?)(?:[-;:&=+$,\\w]+@)?' +
+    '[A-Za-z0-9.-]+|(?:www.|[-;:&=+$,\\w]+@)[A-Za-z0-9.-]+)' +
+    '((?:\\/[\\+~%\\/.\\w-_]*)?\\??(?:[-+=&;%@.\\w_]*)' +
+    '#?(?:[.!\\/\\w]*))|^[A-Za-z0-9_\\/-]+\\.\\w{2,4})');
+  var rClass = /^[_a-zA-Z0-9-]+$/;
+  var rSize = /^\d+(?:|\.\d+)(?:px|%)?$/;
+  var captionClass = 'caption';
+  var noCaptionClass = 'nocaption';
+  var clearClass = 'clear';
+  var autoplayClass = 'autoplay';
+  var loopClass = 'loop';
+  var mutedClass = 'muted';
+  var noControlsClass = 'nocontrols';
+  /**
+   * Video tag
+   *
+   * Syntax:
+   *     {% video [classes] /path/to/video
+   *     [width%] [title text] %}
+   * E.g:
+   *     {% video http://example.com/video145.mp4
+   *     100% "A beautiful sunrise" %}
+   */
+  hexo.extend.tag.register('video', function(args) {
+    var original;
+    var poster = '';
+    var width = '';
+    var height = '';
+    var classes = [];
+    var html = '';
+    var clear = '';
+    // Get CSS classes
+    while (args.length && rClass.test(args[0])) {
+      classes.push(args.shift());
+    }
+    
+    // Get path of original video
+    original = args.shift();
+
+    // Get path of poster image
+    if (args.length && rPath.test(args[0])) {
+      poster = args.shift();
+    }
+    
+    // Get width of video
+    if (args.length && rSize.test(args[0])) {
+      width = args.shift();
+    }
+    
+    // Get height of video
+    if (args.length && rSize.test(args[0])) {
+      height = args.shift();
+    }
+    
+    // Get title of video
+    var title = args.join(' ');
+
+    // Build the video HTML structure
+    var video = '<video';
```

Comment:
I suggest we add an extra space here '<video ' instead of adding this one with the next statement because it's easier to follow and already valid. I suggest we do the same for all other concerned statements.